app/views/topics/index.html.erb

```diff
@@ -9,7 +9,7 @@
     </tr>
     <% @topics.each do |topic| %>
     <tr>
-      <td><%= topic.name %></td>
+      <td> <%= link_to topic.name, topic_path(topic.id) %> </td>
```

Comment:
With rails path and url helpers you don't need to extract the id from the topic. The following should just work:
```
topic_path(topic)
```